### DISCUSSION TITLE
Problems in model running after adding spatial_file

### DISCUSSION THREAD

**Discussion by azsfjeiwfweiofgefg43** [NCAR/hrldas#173] [Q&A]
Hello,

I run into a problem after modifying "soil_data_option", namely adding spatial_file to the namelist.hrldas. Previously, I ran the model successfully without this file. The error is "SNOW HEIGHT NOT FOUND - VALUE DEFINED IN LSMINIT", and I have tracked back the source code where the program crashed as follows. Can anyone help solve this problem? Thank you in advance.

![question](https://github.com/NCAR/hrldas/assets/165929310/e57aaa2d-fb9d-4a99-90ac-1d26f1057000)
![SharedScreenshot](https://github.com/NCAR/hrldas/assets/165929310/7bd27075-9b54-474a-81ec-e19d0eeea8ce)

**Comment by cenlinhe** [comment]
The "SNOW HEIGHT" is not the reason for model crash. The snow height if not provided in the input data file will be automatically initialized in NoahMP initialization module.
Because there is no further error message here, I would guess something might be wrong with the input spatial soil data file. 
which soil data option are you using? 
if you use soil_data_option = 2, please check this I/O module to see if you include all the info in your input soil file: https://github.com/NCAR/hrldas/blob/master/hrldas/IO_code/module_hrldas_netcdf_io.F#L1374-L1471
if you use soil_data_option = 3, please check this I/O module to see if you include all the info in your input soil file: https://github.com/NCAR/hrldas/blob/master/hrldas/IO_code/module_hrldas_netcdf_io.F#L1312-L1370
if you use soil_data_option = 4, please check this I/O module to see if you include all the info in your input soil file: https://github.com/NCAR/hrldas/blob/master/hrldas/IO_code/module_hrldas_netcdf_io.F#L865-L1308
